frontend/src/metabase/query_builder/actions/object-detail.js

```diff
@@ -56,13 +57,34 @@ export const followForeignKey = createThunkAction(
         return false;
       }
 
-      const newCard = startNewCard("query", card.dataset_query.database);
-
-      newCard.dataset_query.query["source-table"] = fk.origin.table.id;
-      newCard.dataset_query.query.filter = getFilterForFK(objectId, fk);
+      const databaseId = card.dataset_query.database;
```

Comment:
Could you create a question from `card` and get `databaseId` this way? We shouldn't be parsing `dataset_query` like this

Reply:
47007574d8792fe8ca40ee59ce4102c97f85f985